articles/how-to-build-a-music-player-using-django/index.md

```diff
@@ -0,0 +1,364 @@
+
+The purpose of this article is to introduce the reader to the Django web framework and explore its ORM and MVT capabilities by engaging him/her in making use of Django to build a simple music player for their personal use.
+
+### Django ORM
+
+Object-Relational Mapping (ORM) is a method that helps you to question and control statistics from a database, the usage of an object-orientated paradigm. When speaking about ORM, we are referring to a library that implements the Object-Relational Mapping technique, the phrase "an ORM".
+
+An ORM library is an everyday library written in your language of desire that encapsulates the code required to control the data so that you do not use raw SQL queries anymore. You engage at once with an object within the same language you are utilizing.
+
+### MVT Architecture
+
+MVT (Model View Template) is a software program layout sample that's a group of 3 elements: Model, View and Template. The Model allows dealing with the database. It is a data access layer that handles the information in the database.
+
+The Template is a presentation layer that handles all the User Interface parts. The View executes the logic and interact with the model to carry data and renders the template.
+
+Although Django follows the MVC pattern, it however, continues its conventions so control is taken care of through the framework itself.
+
+There isn't any separate controller and the entire framework is primarily based totally on Model, View and Template. That's why it's largely known as the MVT framework.
+
+In the MVC architecture, a user sends a request for a resource to Django, Django works as a controller and check for the available resource in the URL.
+
+If URL maps, a view is called that interact with model and template, it renders a template.
+
+Django responds to the user and sends a template as a response.
+
+The principal distinction among the 2 styles is that Django itself looks after the Controller part (Software Code that controls the interactions among the Model and View), leaving us with the template. The template is an HTML record combined with Django Template Language (DTL).
+
+Here is an easy diagram that indicates the MVT structure in Django:
+
+![mvt structure](/engineering-education/how-to-build-a-music-player-using-django/mvt.png)
+
+[Image source](https://www.javatpoint.com/django/images/django-mvt-based-control-flow.png)
+
+### Getting Started with Django
+
+If you are new to Django and want to get started with it, you can refer to the [official Django documentation](https://docs.djangoproject.com/en/3.1/).
+
+### Prerequisites
+
+Install python and enable `pip`. Then run the following to install the required packages:
+
+```bash
+pip install django
+pip install pillow
+```
+
```

Comment:
Since you are not explaining every piece of code, please add the required skill level of the reader to go through this article.